update-result.js

```diff
@@ -0,0 +1,161 @@
+"use strict";
+
+var fs = require("fs");
+var path = require("path");
+var acorn = require("acorn");
+
+/**
+ * Parses a `data-` file, updates/inserts a new value eg "resultKey: result" into the named test's `res` property,
+ * and saves the file in place, ready to be reviewed and committed.
+ *
+ * Usage:
+ * updateResult("es5", ["Root test name", "Subtest name"], "graalvm19", "true");
+ *
+ * @param {string} suite Suite name corresponding to results in ./data-{suite}.js
+ * @param {string[]} testPath Array of test names, from top-level to the test to update
+ * @param {string} resultKey Key for the engine/runtime under test. Alphanumeric + underscores
+ * @param {string} result The *raw* string to insert as the value - eg for a fail, use 'false'
+ */
+function updateResult(suite, testPath, resultKey, result) {
+  var suiteFile = path.join(__dirname, "data-" + suite + ".js");
+  var srcString = fs.readFileSync(suiteFile, "utf8");
+  var testNodes = parseSourceForRootTestNodes(srcString);
+  var testNode;
+  var testNameNode;
+  var readableTestPath = [];
+  for (var i = 0; i < testPath.length; i++) {
+    if (!testNodes || testNodes.type !== "ArrayExpression") {
+      throw new Error(
+        "Expected subtests at " +
+          suite +
+          ', test "' +
+          testPath.slice(0, i + 1).join('" / "') +
+          '"'
+      );
+    }
+    testNode = testNodes.elements.find(function (testNode) {
+      return getTestName(testNode) === testPath[i];
+    });
+    if (!testNode) {
+      throw new Error(
+        'No subtest with name "' +
+          testPath[i] +
+          '" in ' +
+          suite +
+          ', test "' +
+          testPath.slice(0, i + 1).join('" / "') +
+          '"'
+      );
+    }
+    testNameNode = maybeGetPropertyValue(testNode, "name");
+    readableTestPath.push(
+      testNameNode ? testNameNode.value : ["MISSING TEST NAME"]
+    );
+    testNodes = maybeGetPropertyValue(testNode, "subtests");
+  }
+  var resNode = maybeGetPropertyValue(testNode, "res");
+  if (!resNode) {
+    throw new Error(
+      'Missing "res" property on "' + readableTestPath.join('" / "') + '"'
+    );
+  }
+
+  // Determine the indentation of the line the object begins (ie, left of "{")
+  var startOfLine = srcString.lastIndexOf("\n", resNode.start) + 1;
+  var indentLength = 0;
+  while (srcString[startOfLine + indentLength] === " ") {
+    indentLength++;
+  }
+
+  var newRawResString = generateNewRawResString(
+    srcString,
+    indentLength,
+    resNode,
+    resultKey,
+    result
+  );
+  fs.writeFileSync(
+    suiteFile,
+    srcString.substring(0, resNode.start) +
+      newRawResString +
+      srcString.substring(resNode.end)
```

Comment:
```suggestion
    srcString.slice(0, resNode.start) +
      newRawResString +
      srcString.slice(resNode.end)
```
same throughout; slice is preferred over substr/substring.